AutomatedTesting/Gem/PythonTests/Atom/tests/MaterialEditor_Atom_BasicTests.py

```diff
@@ -0,0 +1,135 @@
+        Report.result(
+            Tests.close_opened_asset,
+            atom_tools_utils.is_document_open(material_asset_file_document_id) is False)
+
+        # 3. Open multiple material asset files then use the CloseAllDocuments bus call to close them all.
+        for material in [test_material_1, test_material_2, test_material_3]:
+            atom_tools_utils.open_document(os.path.join(test_data_path, material))
+        Report.result(
+            Tests.close_all_opened_assets,
+            atom_tools_utils.close_all_documents() is True)
+
+        # 4. Open multiple material asset files then use the CloseAllDocumentsExcept bus call to close all but one.
+        test_material_1_document_id = atom_tools_utils.open_document(os.path.join(test_data_path, test_material_1))
+        test_material_2_document_id = atom_tools_utils.open_document(os.path.join(test_data_path, test_material_2))
+        test_material_3_document_id = atom_tools_utils.open_document(os.path.join(test_data_path, test_material_3))
+        atom_tools_utils.close_all_except_selected(test_material_1_document_id)
+        Report.result(
+            Tests.close_all_opened_assets_except_one,
+            atom_tools_utils.is_document_open(test_material_1_document_id) is True and
+            atom_tools_utils.is_document_open(test_material_2_document_id) is False and
+            atom_tools_utils.is_document_open(test_material_3_document_id) is False)
+
+        # 5. Verify Material Asset Browser pane visibility.
+        atom_tools_utils.set_pane_visibility("Asset Browser", True)
+        Report.result(
+            Tests.inspector_pane_visible,
+            atom_tools_utils.is_pane_visible("Asset Browser") is True)
+
+        # 6. Change the baseColor.color property of the test_material_1 material asset file document.
+        base_color_property_name = "baseColor.color"
+        document_id = atom_tools_utils.open_document(os.path.join(test_data_path, test_material_1))
+        initial_color = material_editor_utils.get_property(document_id, base_color_property_name)
+        expected_color = math.Color(0.25, 0.25, 0.25, 1.0)
+        material_editor_utils.set_property(document_id, base_color_property_name, expected_color)
+        Report.result(
+            Tests.changed_material_asset_color,
+            material_editor_utils.get_property(document_id, base_color_property_name) == expected_color)
+
+        # 7. Revert the baseColor.color property of the test_material_1 material asset file document.
+        material_editor_utils.set_property(document_id, base_color_property_name, initial_color)
+        Report.result(
+            Tests.reverted_material_asset_color_change,
+            material_editor_utils.get_property(document_id, base_color_property_name) == initial_color)
+
+        # 8. Look for errors and asserts.
+        TestHelper.wait_for_condition(lambda: error_tracer.has_errors or error_tracer.has_asserts, 1.0)
```

Comment:
Is Tracer guaranteed to update these values asynchronously? Have you proved that this pattern can detect errors/asserts that get added between sleeps?

Not a test structure issue as this is only reporting debugging info on what it sees, not and not setting pass/fail results.

Reply:
I believe it does and we had plans to post this error/assert info in a separate ticket. If you reach out to me in DM I can link you the internal tickets pertaining to that. We just haven't pursued it yet (other priorities)